trigger-up.sh

```diff
@@ -19,20 +19,25 @@ vagrant ssh-config "$BOX_NAME" > "$HOME/.ssh/config.d/${BOX_NAME}"
 SSH_CONFIG_HAS_INCLUDE_D="$(grep -i 'Include config.d/*' "$HOME/.ssh/config" || true )"
 
 if [ -z "$SSH_CONFIG_HAS_INCLUDE_D" ]; then
+    echo "Backup $HOME/.ssh/config"
+    cp $HOME/.ssh/config $HOME/.ssh/config.bak
     echo "Adding 'Include config.d/*' to '$HOME/.ssh/config' file..."
-
-    # Define newline. OSX sed does not know how to interpret \n
-    # character. This must be spread over two lines and must not
-    # include spaces!
-    nl='
-'
-    # Prepend content to the start of the file
-    sed -i '1s;^;Include config.d/*'"\\${nl}\\${nl}"';' "$HOME/.ssh/config"
-    unset nl
+    printf '%s\n%s\n' "Include config.d/*" "" "$(cat $HOME/.ssh/config)" > $HOME/.ssh/config
 else
     echo "No need to modify '$HOME/.ssh/config'. Continuing..."
 fi
 
+HOSTS_HAS_BOX_NAME="$(grep -i $BOX_NAME "/etc/hosts" || true )"
+
+if [ -z "$HOSTS_HAS_BOX_NAME" ]; then
+    echo "Backup /etc/hosts"
+    sudo cp /etc/hosts /etc/hosts.bak 
+    echo "Adding '$BOX_NAME' to /etc/hosts file on local loopback"
+    sudo bash -c "echo '127.0.0.1 $BOX_NAME' >> /etc/hosts"
```

Comment:
My understanding is that while using docker _inside_ Vagrant-managed VM (either via Parallels or some other provider), the VM will run in a subnet that is strictly **not** loopback, and thus it will have IP address from one of [private address ranges defined in RFC1918](https://datatracker.ietf.org/doc/html/rfc1918#section-3).

For that reason, I believe this hosts entry would _normally_ cause issues. Also, I know this is not the valid argument, but just to let you know that for me it works without ever touching /hosts file.

Could you please shed some light on why is this modification of hosts file necessary?

Reply:
I did some testing on my end to see what happens, and indeed, it appears that Vagrant is properly managing the `/etc/hosts` file without any additional configuration.

Steps I took to test it:

1. Confirm that there is nothing related to vagrant VM in `/etc/hosts` file
2. Run `vagrant up`
3. Check `/etc/hosts` file. It should contain line such as `10.x.y.z workbox.shared workboxbox #prl_hostonly shared`
4. Run `vagrant halt`
5. Check `/etc/hosts` file. The line from step 3 is now gone. The `/etc/hosts` file is at the same state as it was on step 1.

@slavekjurkowski2 which version of Vagrant are you running?